### DISCUSSION TITLE
Any interest in other platform SDK, Swift or Kotlin for example

### DISCUSSION THREAD

**Discussion by Stegnerd** [pocketbase/pocketbase#391] [Q&A] [answered]
I was just curious and this project seems amazing! Would there be any interest in creating another sdk? I saw that there was js and dart and I don't use either of those sadly. I think it would be a fun exercise to make another one! No worries if not, i'll look for a task here to contribute to :)  Sorry if that has been asked before but I couldn't find a discussion about it.

**Comment by ganigeorgiev** [comment, marked as answer]
For now I'm not planning to work on SDKs for Kotlin or Swift. Community contributions are welcomed.

If you want to work on this, you can check how the [Dart SDK](https://github.com/pocketbase/dart-sdk) is structured and create a similar API for Kotlin/Swift (_as long as it feels idiomatic for the specific language_).

**Reply by jacobhq** [reply]
I'd be happy to work on a sort of community Rust SDK as I use Rust and love this project. I have seen a few issues relating to it.

**Comment by Stegnerd** [comment]
awesome :) that was my plan to look at the dart one! tysm!

**Reply by ganigeorgiev** [reply]
Please keep in mind that there are planned breaking changes for the `users` and `records` services in an attempt to simplify the users management per https://github.com/pocketbase/pocketbase/issues/376.

**Comment by BasToTheMax** [comment]
What about a Java SDK?

**Reply by ganigeorgiev** [reply]
I'm not planning to work on new SDKs for now, at least not until we reach a little more stable version (you could check the roadmap here - https://github.com/orgs/pocketbase/projects/2).

Community contributions are always welcomed (_please note that currently there are expected to be some small breaking changes in relation to the ongoing users managenement refactoring in #376_).

**Comment by briannadoubt** [comment]
I've put together the beginnings of a swift client! https://github.com/briannadoubt/PocketBase
It still needs auth some auth tunings and realtime->subscribe is throwing me a 403, but I've gotten it to be pretty simple!
```swift
import PocketBase
import SwiftUI

@main
struct PBTestApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

struct Test: Codable, Identifiable {
    var id: String?
    var collectionId: String
    var collectionName: String
    var created: String
    var updated: String
    var foo: String
    enum CodingKeys: String, CodingKey {
        case id
        case collectionId = "@collectionId"
        case collectionName = "@collectionName"
        case created
        case updated
        case foo
    }
}

struct ContentView: View {
    
    @Query("test") var test: [Test]
    
    var body: some View {
        NavigationView {
            List {
                ForEach(test) { test in
                    Text(test.foo)
                }
                if test.isEmpty {
                    Text("No test objects available")
                }
            }
            .refreshable(action: $test.refresh)
            .navigationTitle("Test Collection")
        }
    }
}
```

**Reply by ganigeorgiev** [reply]
> realtime->subscribe is throwing me a 403

We are throwing 403 on `POST /api/realtime` only if the first authenticated client auth model differ from the auth model submitted with the latest subscribe call for the same client. Or in other words:

```
// ✓ initial subscription as guest (aka. the request doesn't have Authorization header)
[guest] POST /api/realtime for clientId1

// ✓ guest -> user1 upgrade (aka. the request has Authorization header)
[user1] POST /api/realtime for clientId1

// x changing the previously authenticated auth model for the same client is not allowed
[user2] POST /api/realtime for clientId1

// x once authenticated, any follow up subscribe requests must also include the Authorization header for user1
[guest] POST /api/realtime for clientId1
```

**Reply by TheFinancialPathfinder** [reply]
> I've put together the beginnings of a swift client! https://github.com/briannadoubt/PocketBase It still needs auth some auth tunings and realtime->subscribe is throwing me a 403, but I've gotten it to be pretty simple!
> 
> ```swift
> import PocketBase
> import SwiftUI
> 
> @main
> struct PBTestApp: App {
>     var body: some Scene {
>         WindowGroup {
>             ContentView()
>         }
>     }
> }
> 
> struct Test: Codable, Identifiable {
>     var id: String?
>     var collectionId: String
>     var collectionName: String
>     var created: String
>     var updated: String
>     var foo: String
>     enum CodingKeys: String, CodingKey {
>         case id
>         case collectionId = "@collectionId"
>         case collectionName = "@collectionName"
>         case created
>         case updated
>         case foo
>     }
> }
> 
> struct ContentView: View {
>     
>     @Query("test") var test: [Test]
>     
>     var body: some View {
>         NavigationView {
>             List {
>                 ForEach(test) { test in
>                     Text(test.foo)
>                 }
>                 if test.isEmpty {
>                     Text("No test objects available")
>                 }
>             }
>             .refreshable(action: $test.refresh)
>             .navigationTitle("Test Collection")
>         }
>     }
> }
> ```

Is this made into a package, or any plans to make it into one?
